### ISSUE TITLE
RADAR like detection ability instead of knowing everything

### ISSUE THREAD

**Issue by Tim2othy** [Tim2othy/spacegame#44]
Give ships the ability to scan space using RADAR/LIDAR. 

Simplest form would be that the minimap doesn't show enemy ships per default. But at any point in time the player can press a key to scan the surrounding area. Then, all ships, in some large radius that aren't behind a planet or asteroid, are marked on the minimap. 

But they all also know where you are.

If we implement this feature, it should also be integrated into the enemy AI #10. So, they only know where the player is once they are close or have used their radar/the player used his radar.

**Comment by lumi-a** [comment]
Rather than having the player press a key, could we just have it running the entire time? That'd be less annoying.

We could add a radar that periodically sweeps the surrounding area, yeah. For a lower-hanging fruit, we could just make the minimap cover a smaller area than the entire universe.

**Comment by Tim2othy** [comment]
yeah, just having a smaller minimap, or enemies just being shown on the minimap if they are in a certain radius of the player seems good enough.

**Comment by lumi-a** [comment]
If we restrict the size of the minimap, we should focus its center on the player. However, that's not possible in splitscreen-mode. If, instead, we focus on an area encompassing all players, with minimum area-diameter 5000 (or something), the players could also exploit this: They could move to opposite corners of the universe, which forces the minimap to focus on the area encompassing them, i.e. the entire universe.

How do you suggest we solve this? I'm not opposed to deleting the minimap entirely because it pollutes the main-loop and has a high performance-drain. Maybe a ship could get a small arrow attached to it that tells it the location of the nearest enemy (calculation of the nearest enemy probably won't be *that* expensive).

**Comment by lumi-a** [comment]
I removed the minimap in #83 (commit 2ec73c6, to be exact), citing my reason as:

> The universe is unbounded now, and I just don't want to deal with all this for now, I'm sorry.

I guess it was too hard to make minimaps work with multiplayer: Because the universe is unbounded now, the minimap can't show "the whole universe", so a minimap should probably chase a player. But which player? Does every player get an own minimap? Large performance drain, and gives me a headache.

I'm fine with having the minimap stay permanently removed.

**Comment by Tim2othy** [comment]
The minimap can't stay permanently removed, then one will just fly into planets all the time and die.

**Comment by Tim2othy** [comment]
I don't think multiplayer is that important so I'd be in favor of the minimap just chasing player 1.
